### ISSUE TITLE
 yarn test command fails 

### ISSUE THREAD

**Issue by farmisen** [prescottprue/generator-react-firebase#170]
**Describe the bug**
Jest tests do not run on a freshly generated app. 

**To Reproduce**

1. Generate a new app with the following options:
```
{
  "generator-react-firebase": {
    "promptValues": {
      "githubUser": "farmisen",
      "firebaseName": "the-keymaster",
      "firebaseKey": "xxxxxxxxxxxxxxxxx",
      "includeRedux": true,
      "includeFirestore": true,
      "otherFeatures": [
        "Continuous Integration config",
        "Firebase Cloud Messaging",
        "Firebase Functions (with ESNext support)",
        "Tests for React Components (Jest)",
        "Firebase Analytics",
        "Error Tracking (Sentry.io)",
        "Client Side Error Reporting (Stackdriver)"
      ],
      "messagingSenderId": "xxxxxxxxxxxxxxxxx",
      "measurementId": "G-xxxxxxxxxxxxxxxxx",
      "appId": "xxxxxxxxxxxxxxxxx",
      "firebasePublicVapidKey": xxxxxxxxxxxxxxxxx",
      "ciProvider": "githubActions",
      "deployTo": "firebase"
    }
  }
}
```
1. run `yarn test`
1. See error:
```
Out of the box, Create React App only supports overriding these Jest options:

  • collectCoverageFrom
  • coverageReporters
  • coverageThreshold
  • coveragePathIgnorePatterns
  • extraGlobals
  • globalSetup
  • globalTeardown
  • moduleNameMapper
  • resetMocks
  • resetModules
  • snapshotSerializers
  • transform
  • transformIgnorePatterns
  • watchPathIgnorePatterns.

These options in your package.json Jest configuration are not currently supported by Create React App:

  • resolver
  • setupFiles
  • testMatch
  • testEnvironment
  • testURL
  • moduleFileExtensions

If you wish to override other Jest options, you need to eject from the default setup. You can do so by running npm run eject but remember that this is a one-way operation. You may also file an issue with Create React App to discuss supporting more options out of the box.
```

**Expected behavior/code**
I should not have to eject the app to run the tests.

**Possible Solution**
Modifying the generator to use `react-app-rewired` to configure jest

**Comment by prescottprue** [comment]
Thanks for posting! Noticed this a while back, and actually had to use [customize-cra](https://github.com/arackaf/customize-cra) along side [react-app-rewired](https://github.com/timarney/react-app-rewired/) to support v3 react-scripts, so it should be easy enough to work that in as an option

**Comment by farmisen** [comment]
Hey @prescottprue I know you! We met at Mel's when he was living in the mission may be 3-4y ago and hacked a little bit of react together. Who would had known back then- I mean, may be beside you - that firebase would have gone such a long way? Never worked on yo generators but happy to send a PR if you point me in the right direction. (Fixing it on my generated app was super easy). I also noticed a few problems with cypress but I'll keep that for later.

**Comment by prescottprue** [comment]
@farmisen Hey, yeah, I remember! Small world! Great to hear from you

The [`generators/app`](https://github.com/prescottprue/generator-react-firebase/tree/master/generators/app) folder contains all of the files for generating the initial app, the other folders in the `generators` folder are "sub-generators". The file templates live in [`generators/app/templates`](https://github.com/prescottprue/generator-react-firebase/tree/master/generators/app/templates) - fitting since a number of the files in the folder include template strings for deciding which code snippets to use depending on options selected by the generator user. [`generators/app/index.js`](https://github.com/prescottprue/generator-react-firebase/blob/master/generators/app/index.js) is where the logic lives to decide which template files to are included.

So in the case of adding an extra config file - we would add the file to `templates` then modify [the logic for when component tests are included](https://github.com/prescottprue/generator-react-firebase/blob/master/generators/app/index.js#L399) to also copy that file. We will also probably change up the [package.json file template](https://github.com/prescottprue/generator-react-firebase/blob/master/generators/app/templates/_package.json).

Feel free to hit me up [over gitter](https://gitter.im/redux-firebase/Lobby) if you want to chat more directly

**Comment by farmisen** [comment]
work is super slow today, giving it a shot

**Comment by prescottprue** [comment]
Fix was included in [the `v5.1.0` release](https://github.com/prescottprue/generator-react-firebase/releases/tag/v5.1.0). I was actually able to remove that config all together and use the version of jest directly within react-scripts, so it also removes some dependencies 🕺 

Reach out if it doesn't work as expected, thanks for reporting!
